src/cr/cube/matrix/cubemeasure.py

```diff
@@ -166,6 +171,72 @@ def means(self):
         return self._means[:, 0, :, 0]
 
 
+# === OVERLAPS ===
+
+
+class _BaseCubeOverlaps(_BaseCubeMeasure):
+    """Base class for overlap cube-measure variants."""
+
+    def __init__(self, dimensions, overlaps, valid_overlaps):
+        super(_BaseCubeOverlaps, self).__init__(dimensions)
+        self._overlaps = overlaps
+        self._valid_overlaps = valid_overlaps
+
+    @classmethod
+    def factory(cls, cube, dimensions, slice_idx):
+        """Return _BaseCubeOverlaps subclass instance appropriate to `cube`.
+
+        Raises `ValueError` if the cube-result does not include a cube-means measure.
+        """
+        if cube.overlaps is None:
+            raise ValueError("cube-result does not contain cube-overlaps measure")
+        dimension_types = cube.dimension_types[-2:]
+        CubeOverlapsCls = (
+            _CatXMrOverlaps if dimension_types[-1] == DT.MR else _MrXCatOverlaps
+        )
+        return CubeOverlapsCls(
+            dimensions,
+            cube.overlaps[cls._slice_idx_expr(cube, slice_idx)],
+            cube.valid_overlaps[cls._slice_idx_expr(cube, slice_idx)],
+        )
+
+    @lazyproperty
+    def overlaps(self):
+        """3D np.float64 ndarray of cube overlaps."""
+        raise NotImplementedError(  # pragma: no cover
+            "`%s` must implement `.overlaps`" % type(self).__name__
+        )
+
+
+class _CatXMrOverlaps(_BaseCubeOverlaps):
+    """Overlaps cube-measure for a NOT_MR_X_MR slice."""
+
+    @lazyproperty
+    def overlaps(self):
+        """3D np.float64 ndarray of overlaps for each valid matrix cell."""
+        return self._overlaps[:, :, 0]
+
+    @lazyproperty
+    def valid_overlaps(self):
+        return self._valid_overlaps[:, :, 0:2].sum(axis=2)
+
+
+class _MrXCatOverlaps(_BaseCubeOverlaps):
+    """Overlaps cube-measure for a MR_X_NON_MR slice.
+
+    Note that the columns-dimensions need not actually be CAT.
+    """
+
+    @lazyproperty
+    def overlaps(self):
+        """3D np.float64 ndarray of overlaps for each valid matrix cell."""
+        return self._overlaps[:, 0, :]
+
+    @lazyproperty
+    def valid_overlaps(self):
+        return self._valid_overlaps[:, 0:2, :].sum(axis=1)
+
+
```

Comment:
why we have MrXCat Overlaps case? I don't see any reason to have overlaps with categorical variable on the column, even because overlaps are used for the pariweise significance and we do not have this features on the rows but only on the cols

Reply:
True, we don't need it. I just added it to confirm that we're able to interpret numbers that come in from zz9. I'll delete, and also delete the tests.

Reply:
ok, ditched this